discovery/aws/ec2_test.go

```diff
@@ -0,0 +1,473 @@
+	attachments := make([]*ec2.InstanceNetworkInterfaceAttachment, 4)
+	enis := make([]*ec2.InstanceNetworkInterface, 4)
+	ips := make([][]*ec2.InstanceIpv6Address, 4)
+
+	// interface without primary IPv6
+	attachments[0] = &ec2.InstanceNetworkInterfaceAttachment{}
+	attachments[0].SetDeviceIndex(3)
+
+	ips[0] = make([]*ec2.InstanceIpv6Address, 1)
+	ips[0][0] = &ec2.InstanceIpv6Address{}
+	ips[0][0].SetIpv6Address(randIpv6())
+	ips[0][0].SetIsPrimaryIpv6(false)
+
+	enis[0] = &ec2.InstanceNetworkInterface{}
+	enis[0].SetAttachment(attachments[0])
+	enis[0].SetIpv6Addresses(ips[0])
+	enis[0].SetSubnetId(*ec2Data.instances[0].SubnetId)
+
+	// interface with primary IPv6
+	attachments[1] = &ec2.InstanceNetworkInterfaceAttachment{}
+	attachments[1].SetDeviceIndex(2)
+
+	ips[1] = make([]*ec2.InstanceIpv6Address, 2)
+	ips[1][0] = &ec2.InstanceIpv6Address{}
+	ips[1][0].SetIpv6Address(randIpv6())
+	ips[1][0].SetIsPrimaryIpv6(false)
+	ips[1][1] = &ec2.InstanceIpv6Address{}
+	ips[1][1].SetIpv6Address(randIpv6())
+	ips[1][1].SetIsPrimaryIpv6(true)
+
+	enis[1] = &ec2.InstanceNetworkInterface{}
+	enis[1].SetAttachment(attachments[1])
+	enis[1].SetIpv6Addresses(ips[1])
+	enis[1].SetSubnetId(*ec2Data.instances[0].SubnetId)
+
+	// interface with primary IPv6
+	attachments[2] = &ec2.InstanceNetworkInterfaceAttachment{}
+	attachments[2].SetDeviceIndex(1)
+
+	ips[2] = make([]*ec2.InstanceIpv6Address, 1)
+	ips[2][0] = &ec2.InstanceIpv6Address{}
+	ips[2][0].SetIpv6Address(randIpv6())
+	ips[2][0].SetIsPrimaryIpv6(true)
+
+	enis[2] = &ec2.InstanceNetworkInterface{}
+	enis[2].SetAttachment(attachments[2])
+	enis[2].SetIpv6Addresses(ips[2])
+	enis[2].SetSubnetId(*ec2Data.instances[0].SubnetId)
+
+	// interface without primary IPv6
+	attachments[3] = &ec2.InstanceNetworkInterfaceAttachment{}
+	attachments[3].SetDeviceIndex(0)
+
+	enis[3] = &ec2.InstanceNetworkInterface{}
+	enis[3].SetAttachment(attachments[3])
+	enis[3].SetIpv6Addresses([]*ec2.InstanceIpv6Address{})
+	enis[3].SetSubnetId(*ec2Data.instances[0].SubnetId)
+
+	ec2Data.instances[0].SetNetworkInterfaces(enis)
+
+	labels := generateExpectedLabels(ec2Data, azIdx, d.cfg.Port)
+
+	labels["__meta_ec2_ipv6_addresses"] = model.LabelValue(
+		fmt.Sprintf(",%s,%s,%s,%s,",
+			*ips[0][0].Ipv6Address,
+			*ips[1][0].Ipv6Address,
+			*ips[1][1].Ipv6Address,
+			*ips[2][0].Ipv6Address,
+		),
+	)
+	labels["__meta_ec2_primary_ipv6_addresses"] = model.LabelValue(
+		fmt.Sprintf(",,%s,%s,", *ips[2][0].Ipv6Address, *ips[1][1].Ipv6Address),
+	)
+	labels["__meta_ec2_primary_subnet_id"] = model.LabelValue(*ec2Data.instances[0].SubnetId)
+	labels["__meta_ec2_subnet_id"] = model.LabelValue(fmt.Sprintf(",%s,", ec2Data.azIDs[azIdx]))
+	labels["__meta_ec2_vpc_id"] = model.LabelValue(*ec2Data.instances[0].VpcId)
+
+	expected := make([]*targetgroup.Group, 1)
+	expected[0] = &targetgroup.Group{
+		Source: ec2Data.region,
+	}
+	expected[0].Targets = append(expected[0].Targets, labels)
+
+	return expected
+}
+
+func TestRefresh(t *testing.T) {
```

Comment:
Suggest `TestEC2DiscoveryRefresh`, as a more helpful name when something random fails.

Reply:
Fixed the name.

Are the other names are fine? Don't you see the test file path or something else unique in the output of the Github action?

Reply:
You do, but have to hunt around.  Also if I want to file a bug about this test I would have to copy-paste parts of the path to form a unique name.

In this example, it's somewhat visible in the 3rd line, and also in the 33rd line:
```
--- FAIL: TestStreamReadEndpoint (0.10s)
    read_handler_test.go:296: 
        	Error Trace:	/__w/prometheus/prometheus/storage/remote/read_handler_test.go:296
        	Error:      	Not equal: 
        	            	expected: []*prompb.ChunkedReadResponse{(*prompb.ChunkedReadResponse)(0xc00369fd40), (*prompb.ChunkedReadResponse)(0xc00369fd80), (*prompb.ChunkedReadResponse)(0xc00369fdc0), (*prompb.ChunkedReadResponse)(0xc00369fe00), (*prompb.ChunkedReadResponse)(0xc00369fe40), (*prompb.ChunkedReadResponse)(0xc00369fe80)}
        	            	actual  : []*prompb.ChunkedReadResponse{(*prompb.ChunkedReadResponse)(0xc00369f9c0), (*prompb.ChunkedReadResponse)(0xc00369fa40), (*prompb.ChunkedReadResponse)(0xc00369fac0), (*prompb.ChunkedReadResponse)(0xc00369fb40), (*prompb.ChunkedReadResponse)(0xc00369fbc0), (*prompb.ChunkedReadResponse)(0xc00369fc80)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -162,4 +162,4 @@
        	            	       Type: (prompb.Chunk_Encoding) 1,
        	            	-      Data: ([]uint8) (len=15) {
        	            	-       00000000  00 01 80 f4 ee 06 40 c7  70 00 00 00 00 00 00     |......@.p......|
        	            	+      Data: ([]uint8) (len=14) {
        	            	+       00000000  00 01 80 f4 ee 06 40 c7  70 00 00 00 00 00        |......@.p.....|
        	            	       },
        	            	@@ -344,4 +344,4 @@
        	            	       Type: (prompb.Chunk_Encoding) 1,
        	            	-      Data: ([]uint8) (len=15) {
        	            	-       00000000  00 01 80 e8 dd 0d 40 d7  70 00 00 00 00 00 00     |......@.p......|
        	            	+      Data: ([]uint8) (len=14) {
        	            	+       00000000  00 01 80 e8 dd 0d 40 d7  70 00 00 00 00 00        |......@.p.....|
        	            	       },
        	Test:       	TestStreamReadEndpoint
FAIL
FAIL	github.com/prometheus/prometheus/storage/remote	110.963s
```